### PR TITLE
CI: differentiate Arrow version from Arrow Meta version

### PR DESCRIPTION
**Pull request by rachelcarmena**
## Reason

This is necessary because of the next Arrow release. 

We are using just a single version for both Arrow and Arrow Meta.

## Changes

- Add an additional version in `gradle.properties`: 
```
ARROW_VERSION=0.10.3-SNAPSHOT
```
- Change **Arrow Meta version** to `1.3.70-SNAPSHOT` because of [Kotlin releases](https://github.com/JetBrains/kotlin/releases).
- Dependencies like `arrow-annotations` and `arrow-ank` have the **Arrow version** now
- CI checks cannot do `./gradlew clean build` because it would fail when **release time** (new version): 
    - **Gradle Plugin** version makes reference to **Arrow Meta Compiler Plugin** published version.
- CI checks have a single file to publish artifacts in order to manage the previous dependence.

## Other version changes

- Remove JitPack repository because `kotlin-compile-testing` has been deployed
- Update `kotlin-compile-testing` dependency

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Package definitions
 GROUP=io.arrow-kt
-VERSION_NAME=0.10.3-SNAPSHOT
+VERSION_NAME=1.3.70-SNAPSHOT
```
Comment (i-walker) [inline gradle.properties]: Shouldn't this be 1.3.60-SNAPSHOT, unless we're jumping versions directly?

Comment (rachelcarmena) [reply gradle.properties]: We start with this version though it will change very soon when sync is ready.

Comment (rachelcarmena) [reply gradle.properties]: Thanks for the review @i-walker!!! :+1:

Comment (i-walker) [reply gradle.properties]: I removed my comment after I read the conv. in Slack. My bad :)

Comment (rachelcarmena) [reply gradle.properties]: Don't worry @i-walker , thanks!! :+1:

**Pull request merged**